tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_vector_find.rs

```diff
@@ -0,0 +1,1865 @@
+    let sks = Arc::new(sks);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    let mut rng = thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = unsigned_modulus(cks.parameters().message_modulus(), NB_CTXT as u32);
+
+    executor.setup(&cks, sks.clone());
+
+    // empty LUT test
+    {
+        let expected_degrees = ExpectedDegrees::new(Degree::new(0), NB_CTXT);
+        let expected_noise_level = ExpectedNoiseLevels::new(NoiseLevel::ZERO, NB_CTXT);
+
+        let empty_lut = &[];
+        let inputs = [
+            cks.encrypt(rng.gen_range(0..modulus)),
+            sks.create_trivial_radix(rng.gen_range(0..modulus), NB_CTXT),
+        ];
+        for ct in inputs {
+            let (result, is_ok) = executor.execute((&ct, empty_lut));
+
+            assert!(result.is_trivial());
+            assert!(is_ok.is_trivial());
+
+            expected_degrees.panic_if_any_is_not_equal(&result);
+            expected_noise_level.panic_if_any_is_not_equal(&result);
+            assert_eq!(is_ok.0.degree.get(), 0);
+            assert_eq!(is_ok.0.noise_level.get(), 0);
+
+            assert_eq!(cks.decrypt::<u64>(&result), 0);
+            assert!(!cks.decrypt_bool(&is_ok));
+        }
+    }
+
+    let expected_degrees = ExpectedDegrees::new(
+        Degree::new(cks.parameters().message_modulus().0 - 1),
+        NB_CTXT,
+    );
+    let expected_noise_level = ExpectedNoiseLevels::new(NoiseLevel::NOMINAL, NB_CTXT);
+
+    // We want to split test in half,
+    // one half where the lut contains the clear, the other half where it does not
+    const HALVED_NB_TEST: usize = if NB_TESTS > 1 { NB_TESTS / 2 } else { 0 };
+
+    let mut lut = Vec::with_capacity(modulus as usize);
+    for i in 0..NB_TESTS {
+        lut.clear();
+
+        let clear = rng.gen_range(0..modulus);
+        let num_values = rng.gen_range(1..MAX_VEC_LEN) as usize;
+        let occurrence_count = if i < HALVED_NB_TEST { 0 } else { 1 };
+        let mut unique_numbers =
+            draw_unique_randoms(&mut rng, num_values, clear, occurrence_count, modulus);
+        for input in unique_numbers.drain(..) {
+            let output = rng.gen_range(0..modulus);
+            lut.push((input, output));
+        }
+
+        let ct = cks.encrypt(clear);
+
+        let (expected_result, expected_is_ok) = lut
+            .iter()
+            .find(|(input, _)| *input == clear)
+            .map_or((0, false), |(_, output)| (*output, true));
+
+        let (result, is_ok) = executor.execute((&ct, &lut));
+
+        // If the mapping only contains numbers (output) that needs less than NB_CTXT
+        // blocks, some trivial zeros will be appended
+        expected_degrees.panic_if_any_is_greater(&result);
+        expected_noise_level.panic_if_any_is_greater(&result);
+        //assert_eq!(is_ok.0.degree, Degree::new(1));
+        //assert_eq!(is_ok.0.noise_level, NoiseLevel::NOMINAL);
```

Comment:
comments ? should these be kept as code or discarded ?

Reply:
should be kept